test/api/routes.test.ts

```diff
@@ -0,0 +1,33 @@
+import request from 'supertest';
+import app from '../../api/server';
+import { getHealth } from '../../api/services/healthService';
+import { getLocations } from '../../api/services/locationService';
+import { getSubcategories } from '../../api/services/taxonomyService';
+
+jest.mock('../../api/services/healthService');
+jest.mock('../../api/services/locationService');
+jest.mock('../../api/services/taxonomyService');
+
+describe('Routes', () => {
+  const routes = [
+    { route: '/api/health/status', serviceMethod: getHealth },
+    {
+      route: '/api/location?taxonomyId=Health%20Care&limit=500&pageNumber=1',
+      serviceMethod: getLocations,
+    },
+    { route: '/api/taxonomy/1/children', serviceMethod: getSubcategories },
+  ];
+
+  Promise.all(
+    routes.map(async ({ route, serviceMethod }) => {
+      await it(`${route}`, async () => {
+        const mock = serviceMethod as jest.Mock;
+        // doesn't matter what's returned, as long as it's not empty
+        mock.mockResolvedValue('');
```

Comment:
Same comment as before - this test can be a lot simpler without the mock of the individual service method, that's what the jest.mock function does. It mocks each function on the imported module from the path that you supply. Not trying to nit pick but just trying to reduce the amount of code we have to write. This would be good enough:
```
app.routes.forEach(async route => await request(app).get(route));
```

[Here](https://stackoverflow.com/a/14934933/4161992) is an answer to get the routes - I'm not sure which method we have to use to get all registered routes.